R/htmlwidgets.R

```diff
@@ -269,17 +274,25 @@ shinyRenderWidget <- function(expr, outputFunction, env, quoted) {
         instance$elementId, "\"; Shiny doesn't use them"
       )
     }
-    x <- .subset2(instance, "x")
     deps <- .subset2(instance, "dependencies")
     deps <- lapply(
       htmltools::resolveDependencies(deps),
       shiny::createWebDependency
     )
-    evals = JSEvals(x)
-    list(x = x, evals = evals, deps = deps)
+    payload = modifyList(createPayload(instance), list(deps = deps))
   }
 
   # mark it with the output function so we can use it in Rmd files
   shiny::markRenderFunction(outputFunction, renderFunc)
 }
 
+# Helper function to create payload
+createPayload <- function(instance){
+  if (!is.null(instance$preRenderHook)){
+    instance <- instance$preRenderHook(instance)
```

Comment:
I think it makes a lot of sense @yihui. I can't see any reasons for the hook function to be carried along, once it has been applied, since there is no use for it in the json.